File: javascript/high-scores/high-scores.js
export class HighScores {
  constructor(scores) {
    this._scores = scores;
  }

  get scores() {
    return this._scores;
  }

  get latest() {
    return this._scores[this._scores.length - 1];
  }

  get personalBest() {
    return Math.max(...this._scores);
  }

  get personalTopThree() {
    return this._scores.sort((score1, score2) => score2 - score1).slice(0, 3);
  }
}

File: cpp/high-scores/high_scores.cpp
#include "high_scores.h"

#include <algorithm>

namespace arcade {

    std::vector<int> HighScores::list_scores() {
        return scores;
    }

    int HighScores::latest_score() {
        return scores.back();
    }

    int HighScores::personal_best() {
        return *max_element(scores.begin(), scores.end());
    }

    std::vector<int> HighScores::top_three() {
        std::vector<int> best = scores;
        sort(best.begin(), best.end(), std::greater<>());
        return {best.begin(), best.begin() + std::min(3, int(best.size()))};
    }

}  // namespace arcade
